### PR TITLE
tkcairo backend

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

same as wxcairo and qtcairo, but for tk :-)

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by tacaswell** [comment]
This did not get in under the RC, but for completeness I think we should still put it in 2.2

**Comment by anntzer** [comment]
Thanks, got a bit delayed by real life and was also waiting for the other backends to get in first.

A few notes on the implementation:

I moved the common parts to `_backend_tk.py`, which plays a role similar to `backend_qt5.py` (but is private because I don't see a good reason to make it public... but perhaps it should? dunno).  The patch looks enormous because github's UI doesn't seem to detect the move, possibly because `backend_tkagg.py` still exists (but is essentially completely new).

I also deprecated FigureManagerTkAgg and NavigationToolbar2TkAgg, both because FigureManagerTk and NavigationToolbar2Tk are available and because I would argue that the former should actually be accessed by third parties as `from matplotlib.backends.backend_foo import FigureManager` (by the design of backends this will necessarily import the manager class corresponding to the backend); while a similar "single" name is not available for NavigationToolbar2 it could be added as well.  However these deprecations are not strictly necessary and can easily be reverted.

File: lib/matplotlib/backends/backend_tkagg.py [hunk 1]
```diff
@@ -305,780 +18,17 @@ def blit(self, bbox=None):
...
-
-    def set_window_title(self, title):
-        self.window.wm_title(title)
-
-    def full_screen_toggle(self):
-        is_fullscreen = bool(self.window.attributes('-fullscreen'))
-        self.window.attributes('-fullscreen', not is_fullscreen)
-
 
 @cbook.deprecated("2.2")
```
Comment (QuLogic) [inline lib/matplotlib/backends/backend_tkagg.py]: Was this replaced by something in particular?

Comment (anntzer) [reply lib/matplotlib/backends/backend_tkagg.py]: This class was used by the **old** navigationtoolbar (pre-navigationtoolbar2) so basically unused for a very long time. But I moved it to _backend_tk like everyone else for a normal deprecation period.

**Comment by QuLogic** [comment]
You might want to rebase and re-target `v2.2.x` directly so that tests aren't borked.

**Comment by anntzer** [comment]
Rebased.  Before I retarget this: do you want to later "forwardport" from 2.2 to master?

**Comment by QuLogic** [comment]
I expect `v2.2.x` will be merged up to `master` at some point (at least until 2.2.0), until it becomes too difficult with all our Python 3 changes. You'll need to rebase again though if you re-target because you'll get all the Python 3 changes in now.

**Comment by anntzer** [comment]
done

**Comment by anntzer** [comment]
needs to be merged into master as well.

**Comment by ImportanceOfBeingErnest** [comment]
This looks like it will cause chaos when a new version is released out of the development version.

If  `FigureManagerTkAgg` `NavigationToolbar2TkAgg` are meant to disappear, there should be an API change notice.
But I would argue that as long as there is no date/version set for `FigureManagerTkAgg` `NavigationToolbar2TkAgg` to disappear they should still be accessible.  
And in any case the deprecation messages should include what to use instead of `FigureManagerTkAgg` `NavigationToolbar2TkAgg`

**Comment by jenshnielsen** [comment]
@ImportanceOfBeingErnest As far as I can see they are both still there. They are just empty aliases for the General non Agg specific version

**Comment by ImportanceOfBeingErnest** [comment]
@jenshnielsen I wouldn't preclude that you have better eyes than I have, but I don't see them anywhere. 
 And in any case, doing `from matplotlib.backends.backend_tkagg import NavigationToolbar2TkAgg` results in an ImportError on current master.

**Comment by anntzer** [comment]
They were deprecated in 2.2 and removed in https://github.com/matplotlib/matplotlib/pull/10426 (3.0) which is admittedly a shorter cycle than normal, but an API removal notice should be enough.

File: lib/matplotlib/backends/backend_tkagg.py [hunk 2]
```diff
@@ -305,780 +18,17 @@ def blit(self, bbox=None):
...
-            tw.tk.call("::tk::unsupported::MacWindowStyle",
-                       "style", tw._w,
-                       "help", "noActivates")
-        except Tk.TclError:
-            pass
-        label = Tk.Label(tw, text=self.text, justify=Tk.LEFT,
-                         background="#ffffe0", relief=Tk.SOLID, borderwidth=1)
-        label.pack(ipadx=1)
+class FigureManagerTkAgg(FigureManagerTk):
+    pass
```
Comment (jenshnielsen) [inline lib/matplotlib/backends/backend_tkagg.py]: @ImportanceOfBeingErnest here is the figure manager

**Pull request merged**